e2e/playwright/actions/src/tests/share/share-file.spec.ts

```diff
@@ -0,0 +1,246 @@
+        expect(await personalFiles.dataTable.performActionFromExpandableMenu(file3, 'Share'));
+        const labels = await personalFiles.shareDialog.getLabels();
+        expect(await personalFiles.shareDialog.getDialogTitle()).toEqual(`Share ${file3}`);
+        expect(labels[0].trim()).toBe(`Share ${file3}`);
+        expect(await personalFiles.shareDialog.getInfoText()).toEqual('Share Link');
+        expect(await personalFiles.shareDialog.getLinkUrl()).toContain(shareLinkPreUrl);
+        expect(await personalFiles.shareDialog.isUrlReadOnly()).toBe(true);
+        expect(await personalFiles.shareDialog.isShareToggleChecked()).toBe(true);
+        expect(labels[1].trim()).toBe('Link Expiry Date');
+        expect(await personalFiles.shareDialog.isExpireToggleEnabled()).toBe(false);
+        expect(await personalFiles.shareDialog.isCloseEnabled()).toBe(true);
+      });
+
+      test('[C286329] Share a file', async ({ personalFiles, nodesApiAction }) => {
+        expect(await personalFiles.dataTable.performActionFromExpandableMenu(file3, 'Share'));
+
+        const url = await personalFiles.shareDialog.getLinkUrl();
+        await personalFiles.shareDialog.clickClose();
+
+        const sharedId = await nodesApiAction.getSharedId(file3Id);
+        expect(url).toContain(sharedId);
+      });
+
+      test('[C286330] Copy shared file URL', async ({ personalFiles, page }) => {
+        expect(await personalFiles.dataTable.performActionFromExpandableMenu(file4, 'Share'));
+
+        const url = await personalFiles.shareDialog.getLinkUrl();
+        expect(url).toContain(shareLinkPreUrl);
+
+        await personalFiles.shareDialog.urlAction.click();
+
+        const shareSnackBar = personalFiles.snackBar;
+        await expect(shareSnackBar.getByMessageLocator('Link copied to the clipboard')).toBeVisible();
+
+        await page.goto(url);
+        await personalFiles.viewer.waitForViewerToOpen();
+
+        const downloadPromise = personalFiles.page.waitForEvent('download');
+        await personalFiles.viewer.toolbar.sharedDownloadButton.click();
+
+        const download = await downloadPromise;
+        expect(download.suggestedFilename()).toBe(file4);
+      });
+
+      test('[C286332] Share a file with expiration date', async ({ personalFiles, nodesApiAction, page }) => {
+        expect(await personalFiles.dataTable.performActionFromExpandableMenu(file5, 'Share'));
+
+        await personalFiles.shareDialog.expireToggle.click();
+        expect(await personalFiles.shareDialog.isExpireToggleEnabled()).toBe(true);
+
+        await personalFiles.shareDialog.datetimePickerButton.click();
+        expect(await personalFiles.shareDialog.dateTimePicker.isCalendarOpen()).toBe(true);
+
+        await personalFiles.shareDialog.dateTimePicker.pickDateTime();
+
+        const inputDate = await personalFiles.shareDialog.getExpireDate();
+
+        await page.waitForTimeout(timeouts.normal);
```

Comment:
can use personalFiles.page

Reply:
waitForTimeout is an inbuilt method of Page Interface provided by Playwright so I guess we don't need personalFiles.page. Otherwise, we will have to implement wateForTimeout for other pages as well which would be redundant.